### ISSUE TITLE
Is this actually related to https://plotly.com/ in any way?

### ISSUE THREAD

**Issue by mavavilj** [SciNim/nim-plotly#80]
Is this actually related to https://plotly.com/ in any way?

**Comment by Vindaar** [comment]
The Nim library part of this only generates the JSON required to draw a plot with `plotly.js`. That part is this file here:

https://github.com/SciNim/nim-plotly/blob/master/src/plotly/api.nim

When viewing such a plot, for example with `show`, an HTML file is generated, which includes a `<script>` tag that loads the JS library. This is done here:

https://github.com/SciNim/nim-plotly/blob/master/src/plotly/tmpl_html.nim#L26

So yes, this library is related to the actual plotly. We're just not in any way an "official" plotly binding.

Does this clear up the confusion?

**Comment by mavavilj** [comment]
> This is not specifically for the javascript nim target (but the javascript target is supported!).

Will this have some more native-ish performance then?

https://cpplot.readthedocs.io/en/latest/about.html

**Comment by mavavilj** [comment]
I'm not sure why people want to use JavaScript for native plotting. Isn't it supposed to be slow as JS?

**Comment by Vindaar** [comment]
> I'm not sure why people want to use JavaScript for native plotting. Isn't it supposed to be slow as JS?

I'm sorry I don't understand what you're saying. Of course if you use the JS backend both the JSON generation as well as plotly.js is "as slow as JS". 

No one uses plotly because they want fast plots. Use something like [ggplotnim](https://github.com/Vindaar/ggplotnim) in case it's important to generate many plots relatively quickly.

**Comment by mavavilj** [comment]
I think my main confusion is because this is listed under:

https://github.com/SciNim

Suggesting that it should be a fast library.

**Comment by mavavilj** [comment]
Yes, but since Nim is in syntax like it, then wouldn't it make more sense to use the native performance on Nim then?
